Extensions/dnSpy.AsmEditor/Bundle/SaveBundleContentsCommands.cs

```diff
@@ -0,0 +1,85 @@
+using dnSpy.AsmEditor.Properties;
+using dnSpy.Contracts.Documents;
+using dnSpy.Contracts.Documents.TreeView;
+using dnSpy.Contracts.Menus;
+using dnSpy.Contracts.TreeView;
+
+namespace dnSpy.AsmEditor.Bundle {
+	sealed class SaveBundleContentsCommand {
+		[ExportMenuItem(Header = "res:SaveBundleContents", Group = MenuConstants.GROUP_CTX_DOCUMENTS_ASMED_BUNDLE, Order = 0)]
+		sealed class DocumentsCommand : DocumentsContextMenuHandler {
+			public override bool IsVisible(AsmEditorContext context) => SaveBundleContentsCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveBundleContentsCommand.Execute(context);
+		}
+
+		[ExportMenuItem(OwnerGuid = MenuConstants.APP_MENU_EDIT_GUID, Header = "res:SaveBundleContents", Group = MenuConstants.GROUP_APP_MENU_EDIT_ASMED_BUNDLE, Order = 0)]
+		sealed class EditMenuCommand : EditMenuHandler {
+			[ImportingConstructor]
+			public EditMenuCommand(IAppService appService) : base(appService.DocumentTreeView) {
+			}
+			public override bool IsVisible(AsmEditorContext context) => SaveBundleContentsCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveBundleContentsCommand.Execute(context);
+		}
+
+		static bool IsSingleFileBundle(AsmEditorContext context) => context.Nodes.Length == 1 && context.Nodes[0] is BundleDocumentNode;
+		static bool CanExecute(AsmEditorContext context) => SaveBundleContentsCommand.IsSingleFileBundle(context);
+		static void Execute(AsmEditorContext context) {
+			var docNode = context.Nodes[0].GetDocumentNode();
+			var bundleDoc = docNode!.Document as DsBundleDocument;
+			Debug2.Assert(bundleDoc != null);
+			Debug2.Assert(bundleDoc.SingleFileBundle != null);
+			SaveBundle.Save(bundleDoc.SingleFileBundle.Entries.ToArray(), dnSpy_AsmEditor_Resources.SaveBundleContents);
+		}
+	}
+
+	sealed class SaveRawEntryCommand {
+		[ExportMenuItem(Header = "res:SaveRawEntry", Group = MenuConstants.GROUP_CTX_DOCUMENTS_ASMED_BUNDLE, Order = 1)]
+		sealed class DocumentsCommand : DocumentsContextMenuHandler {
+			public override bool IsVisible(AsmEditorContext context) => SaveRawEntryCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveRawEntryCommand.Execute(context);
+		}
+
+		[ExportMenuItem(OwnerGuid = MenuConstants.APP_MENU_EDIT_GUID, Header = "res:SaveRawEntry", Group = MenuConstants.GROUP_APP_MENU_EDIT_ASMED_BUNDLE, Order = 1)]
+		sealed class EditMenuCommand : EditMenuHandler {
+			[ImportingConstructor]
+			public EditMenuCommand(IAppService appService) : base(appService.DocumentTreeView) {
+			}
+			public override bool IsVisible(AsmEditorContext context) => SaveRawEntryCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveRawEntryCommand.Execute(context);
+		}
+
+		static bool IsBundleSingleSelection(AsmEditorContext context) => context.Nodes.Length == 1 && context.Nodes[0] is IBundleEntryNode;
```

Comment:
Check whether the `IBundleEntryNode.BundleEntry` is not null. Implementing this interface does not guarantee that the node has an entry!